src/server.c

```diff
@@ -73,6 +73,7 @@ static const struct option options[] = {
     {"ssl-key", required_argument, NULL, 'K'},
     {"ssl-ca", required_argument, NULL, 'A'},
     {"url-arg", no_argument, NULL, 'a'},
+    {"arg-file", required_argument, NULL, 'f'},
```

Comment:
why is this required argument?

Reply:
I believe an optional argument **must** be passed in without a space (`-f{argument}` instead of `-f {argument}`). I figured this might be confusing for users when the rest of the options can be listed as `-{option} {argument}`, especially since using `-f {argument}` would make ttyd try to use this argument as a command instead.

If this isn't an issue, I can change the argument to be optional.

Reply:
TLDR: We can hold off discussion on this until ttyd maintainers respond as i feel this is a matter of taste mostly and we should please the reviewer 😛 


skimmed through https://man7.org/linux/man-pages/man3/getopt.3.html. It seems the for optional arguments `-f  arg` `-f=arg` both should work.

> believe an optional argument must be passed in without a space
have you checked it?


In either case, optional might be the right thing to do. But if its an optional argument is there a way to distinguish between the the user running `ttyd -f mycmd` and `ttyd mycmd`?

IMO Ideally (you might have different thoughts, which is reasonable)

1. `ttyd mycmd` : Our code should not trigger
2. `ttyd -f /tmp/myprefix mycmd` : Our code should trigger and create a tmp file with `/tmp/myprefix` prefix
3. `ttyd -f mycmd` : Our code should trigger and create a tmp file with `/tmp/` prefix (or something like `/tmp/ttyd-args-` prefix)

overall this i feel this is quite opinionated!

Reply:
also as @kahing  was saying in another comment, not supporting the following is quite reasonable

> ttyd -f /tmp/myprefix mycmd : Our code should trigger and create a tmp file with /tmp/myprefix prefix

Reply:
I agree with the cases you have listed there. I've tried making this argument optional but it seems like only cases `1` and `3` work as expected. For `ttyd -f /tmp/myprefix mycmd`, `/tmp/myprefix` isn't being passed in as an optional argument for `-f`, rather it is being used as the ttyd command. 

Running `ttyd -f=/tmp/myprefix mycmd` sets the prefix to `=/tmp/myprefix` and running `ttyd -f/tmp/myprefix mycmd` sets the prefix to `/tmp/myprefix`. 
There's a discussion about this behaviour [here](https://stackoverflow.com/questions/1052746/getopt-does-not-parse-optional-arguments-to-parameters).

If this behaviour isn't too confusing, I can make the argument optional.

Alternatively, we could do something like `-f -prefix /tmp/myprefix` where `-f` has no argument (turns on the url argument to file feature) and `-prefix` has a required argument that specifies a path. Then if only `-f` is specified, we use some default prefix like `/tmp/ttyd-args`.